src/main/resources/templates/java/test/testFiles/structural/StructuralTest.java

```diff
@@ -84,6 +107,39 @@ protected boolean checkModifiers(String[] observedModifiers, JSONArray expectedM
         return allModifiersAreImplemented;
     }
 
+    protected boolean checkAnnotations(Annotation[] observedAnnotations, JSONArray expectedAnnotations) {
+
+        // If both the observed and expected elements have no annotations, then they match.
+        // A note: for technical reasons, we get in case of no observed annotations, a string array with an empty string.
+        if(observedAnnotations.length == 0 && expectedAnnotations.length() == 0) {
+            return true;
+        }
+
+        // If the number of the annotations does not match, then the annotations per se do not match either.
+        if(observedAnnotations.length != expectedAnnotations.length()) {
+            return false;
+        }
+
+        // Otherwise check if each expected annotation is contained in the array of the observed ones.
+        // If at least one isn't, then the modifiers don't match.
+        for(Object expectedAnnotation : expectedAnnotations) {
```

Comment:
This could also be done with streams and/or set difference, e.g. (not tested):

```
        Set<String> observedAnnotationNames = Arrays.stream(observedAnnotations).map(annotation -> annotation.annotationType().getSimpleName()).collect(Collectors.toSet());
        Set<String> expectedAnnotationNames = expectedAnnotations.toList().stream().map(annotation -> (String) annotation).collect(Collectors.toSet());
        expectedAnnotationNames.removeAll(observedAnnotationNames);
        return expectedAnnotationNames.isEmpty();
```